### PR TITLE
analog button service

### PR DESCRIPTION
**Pull request by pelikhan**
Refactoring the button service to support both digital and analog buttons by converting the boolean ``pressed`` reading into "pressure" reading, u0.16

rationale:
- [ ] we want a unified set of events for buttons, whether they are digital/analog
- [ ] users should preferrably use events, which abstract away debouncing, less packets on the bus
- [ ] pressed register was exposing a raw sensor data, which may not completly sync up with the debounced events.
- [ ] pressure button is now a "configuration service" which allows to set the threshold but not read the sensor.
- [ ] capacitive button is now a "configuration server" which allows to set the thresholds, calibrate but not read the sensor.

@jamesadevine this is probably a breaking change for you.

### CODE REVIEW

File: services/capacitivebutton.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    identifier: 0x1865adc9
+    tags: 8bit
+    group: button
+
+A configuration service for a capacitive push-button.
+
+## Registers
+
+    rw down_threshold: u0.16 / @ inactive_threshold
```
Comment (mmoskal) [inline services/capacitivebutton.md:11]: so the idea for up/down is that if the current value is between them, then the button retains previous state? once it crosses into say active range, then after a few frames it becomes active, and we wait until it crosses back into inactive for a few frames to de-activates it?

**Comment by jamesadevine** [comment]
@pelikhan The touch test module is just for testing capacitive touch, so no breaking changes for me.

Does the padauk button implementation need updating before I flash the MCUs @mmoskal ?

**Comment by pelikhan** [comment]
Yes let’s sync

________________________________

File: services/capacitivebutton.md (hunk 1 above)
Comment (pelikhan) [reply services/capacitivebutton.md:11]: right, we only need a active threshold

Comment (pelikhan) [reply services/capacitivebutton.md:11]: CODAL uses a single threshold

Comment (pelikhan) [reply services/capacitivebutton.md:11]: https://github.com/lancaster-university/codal-core/blob/master/source/drivers/TouchButton.cpp

**Comment by mmoskal** [comment]
should we use the 0x2... range for the "mixin" services here? https://github.com/microsoft/jacdac/issues/162

I think we only need to update the spectool to accept that range, otherwise there should be no changes needed.

**Comment by pelikhan** [comment]
@mmoskal https://github.com/microsoft/jacdac/pull/357/commits/abd3ab7f74a042c8b59c6946207c9d13d4d10601

**Review by mmoskal — APPROVED** [review]

**Pull request merged**